### PR TITLE
Deprected explode router

### PR DESCRIPTION
**Pull request by hima-mage**
[php 8.1 - explode(): Passing null to parameter #2 ($string) of type string is deprecated]
so i used null coalescing operator to avoid that error

### CODE REVIEW

File: Controller/Router.php [hunk 1]
```diff
@@ -83,8 +83,8 @@ public function match(RequestInterface $request)
         } elseif (strpos($identifier, $rssAction) !== false) {
             $identifier = $this->checkRssIdentifier($identifier);
         }
-
-        $routePath = explode('/', $identifier);
+ 
+        $ids = explode('/', $identifier ?? '');
```
Comment (mercurieus) [inline Controller/Router.php:87]:
```suggestion
        $routePath = explode('/', $identifier ?? '');
```

File: Block/Listpost.php [hunk 2]
```diff
@@ -46,7 +46,7 @@ public function getPostCollection()
             $pager = $this->getLayout()->createBlock(Pager::class, 'mpblog.post.pager');
 
             $perPageValues = (string)$this->helperData->getConfigGeneral('pagination');
-            $perPageValues = explode(',', $perPageValues);
+            $perPageValues = explode('/', $perPageValues ?? '');
```
Comment (mercurieus) [inline Block/Listpost.php:49]:
```suggestion
            $perPageValues = explode(',', $perPageValues ?? ''); 
```

File: Block/Listpost.php [hunk 3]
```diff
@@ -69,7 +69,7 @@ public function maxShortDescription($description)
     {
         if (is_string($description)) {
             $html = '';
-            foreach (explode("\n", trim($description)) as $value) {
+            foreach (explode('/', trim($description) ?? '') as $value) {
```
Comment (mercurieus) [inline Block/Listpost.php:72]:
```suggestion
            foreach (explode('\n', trim($description) ?? '') as $value) {
```

File: Block/Adminhtml/Category/Tree.php [hunk 4]
```diff
@@ -175,7 +175,7 @@ protected function _getNodeJson($node, $level = 0)
             $node = new Node($node, 'category_id', new \Magento\Framework\Data\Tree());
         }
 
-        $storeIds = $node->getStoreIds() ? explode(',', $node->getStoreIds()) : [];
+        $storeIds = $node->getStoreIds() ? explode('/', $node->getStoreIds() ?? '')   : [];
```
Comment (mercurieus) [inline Block/Adminhtml/Category/Tree.php:178]:
```suggestion
        $storeIds = $node->getStoreIds() ? explode(',', $node->getStoreIds() ?? '')   : [];
```

File: Block/Adminhtml/Post/Edit/Tab/Renderer/Category.php [hunk 5]
```diff
@@ -157,8 +157,8 @@ public function getCategoriesTree()
     public function getValues()
     {
         $values = $this->getValue();
-        if (!is_array($values)) {
-            $values = explode(',', $values);
+        if (!is_array($values)) { 
+            $values = explode('/', $values ?? '');
```
Comment (mercurieus) [inline Block/Adminhtml/Post/Edit/Tab/Renderer/Category.php:161]:
```suggestion
            $values = explode(',', $values ?? '');
```

File: Block/Adminhtml/Post/Edit/Tab/Renderer/Tag.php [hunk 6]
```diff
@@ -145,7 +145,7 @@ public function getValues()
         $values = $this->getValue();
 
         if (!is_array($values)) {
-            $values = explode(',', $values);
+            $values = explode('/', $values ?? '');
```
Comment (mercurieus) [inline Block/Adminhtml/Post/Edit/Tab/Renderer/Tag.php:148]:
```suggestion
            $values = explode(',', $values ?? '');
```

File: Block/Adminhtml/Post/Edit/Tab/Renderer/Topic.php [hunk 7]
```diff
@@ -145,7 +145,7 @@ public function getValues()
         $values = $this->getValue();
 
         if (!is_array($values)) {
-            $values = explode(',', $values);
+            $values = explode('/', $values ?? '');
```
Comment (mercurieus) [inline Block/Adminhtml/Post/Edit/Tab/Renderer/Topic.php:148]:
```suggestion
            $values = explode(',', $values ?? '');
```

File: Controller/Router.php (hunk 1 above)
Comment (mercurieus) [inline Controller/Router.php:87]:
```suggestion
        $routePath = explode('/', $identifier ?? '');
```

**Review by mercurieus — CHANGES_REQUESTED** [review]

**Comment by Leobaillard** [comment]
@zenfork Thanks for the PR, I need this for a project (2.4.4/PHP8.1) and I'm using your PR as a composer patch to make it work. Unfortunately, the suggested change that you made in `Controller/Router.php` by renaming `$ids` to `$routePath` is needed for it to work and it is not included in the Github patch version of the PR. Could you validate your own suggestion so that it can be used while the PR gets approved by the plugin's author? Thanks a lot!

Edit: woops sorry, I meant to thank @ishiekh-dev for the PR! And @zenfork for the suggestions.

**Review by mercurieus — APPROVED** [review]

**Comment by Leobaillard** [comment]
Looks like approving the changes does not change the auto diff patch. @ishiekh-dev could you maybe include the suggestions to your PR so that the diff gets updated? Thanks!

**Comment by Leobaillard** [comment]
Ping @ishiekh-dev ?

**Comment by KhietNguyen83** [comment]
Hi ,
Thanks for your contribution.
We have updated and released a new version that solves this problem.
Thank you very much!

**Pull request closed without merging**